roles/libvirt_manager/tasks/gather_vm_networks.yml

```diff
@@ -0,0 +1,39 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: List the attached interfaces.
+  ansible.builtin.command:
+    cmd: >-
+      virsh -q -c qemu:///system
+      domiflist
+      {{ domain_name }}
+  register: _dom_list
+
+- name: Gather the network and mac address.
+  vars:
+    _values: "{{ item.split() }}"
+    _data:
+      - key: "{{ domain_name | replace('cifmw-', '') }}"
+        value:
+          - network: "{{ _values[2] | replace('cifmw-', '') }}"
+            mac: "{{ _values[4] }}"
+  ansible.builtin.set_fact:
```

Comment:
ok, sooo. If you we go with that approach, we don't want to set_fact for that `cifmw_libvirt_manager_mac_map` in the `deploy_layout.yml` line 108. It doesn't make sense imho.

So here, we'd just  `cifmw_libvirt_manager_mac_map | default({}) | combine(....)` and we're good.

As I understand, this tasks file is called once, and will generate the data once for all for every existing instances. It should be more robust than the previous implementation, while avoiding to do some "`slurp` if file exists" and all.

Reply:
Thanks. It makes sense. Will fix it in the next patch.